auth/src/main/java/org/aerogear/mobile/auth/credentials/OIDCCredentials.java

```diff
@@ -67,71 +62,40 @@ public String getRefreshToken() {
         return authState.getRefreshToken();
     }
 
-    public IntegrityCheckParameters getIntegrityCheckParameters() { return this.integrityCheckParameters; }
-
     /**
-     * Verify the authenticity of a JWT token against integrity parameters (provided config).
-     * @param jwtToken The JWT token to verify.
-     * @return <code>true</code> if the token integrity is good.
-     */
-    public boolean verifyToken(final String jwtToken) {
-        return verifyToken(nonEmpty(jwtToken, "jwtToken"),
-            integrityCheckParameters.getPublicKey(),
-            integrityCheckParameters.getIssuer(),
-            integrityCheckParameters.getAudience());
-    }
-
-    /**
-     * A function to verify the authenticity of a JWT token against a public key, expected issuer and audience.
-     * @param jwtToken - The JWT Token to Verify.
-     * @param publicKey - The Public Key from Keycloak, without the Begin/End tags.
-     * @param issuer - The expected Issuer of the JWT
-     * @param audience - The expected Audience of the JWT
-     * @return <code>true</code> if the token integrity is good.
-     * @throws IllegalArgumentException
+     * Verify the token and its claims against the given Keycloak configuration
+     * @param keycloakConfig
+     * @return
      */
-    public boolean verifyToken(final String jwtToken, final String publicKey, final String issuer, final String audience) {
-        nonEmpty(publicKey, "publicKey");
-        nonEmpty(jwtToken, "jwtToken");
-
-        final String constructedPublicKey = beginPublicKey + publicKey + endPublicKey;
-
-        // Convert the public key from a string to a Java security key
-        final RsaKeyUtil utils = new RsaKeyUtil();
-        final PublicKey jwtPublicKey;
-        try {
-            jwtPublicKey = utils.fromPemEncoded(constructedPublicKey);
-        } catch (JoseException e) {
-            Log.e(TAG, e.getMessage(), e);
-            throw new IllegalArgumentException(e);
-        } catch (InvalidKeySpecException e) {
-            Log.e(TAG, e.getMessage(), e);
-            // If the public key is invalid then we cannot determine the tokens integrity.
-            return false;
-        }
-
-
+    public boolean verifyClaims(final JsonWebKeySet jwks, final KeycloakConfiguration keycloakConfig) {
+        String issuer = keycloakConfig.getHostUrl() + "/realms/secure-app";
```

Comment:
this should be returned from the `keycloakConfig` and it should be in the format of`{keycloakHost}/realms/{realmName}`